independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/CurrentManagedContext.java

```diff
@@ -280,23 +272,39 @@ public Map<InjectableBean<?>, Object> getContextualInstances() {
          * @return {@code true} if the state was successfully invalidated, {@code false} otherwise
          */
         boolean invalidate() {
-            // Atomically sets the value just like AtomicBoolean.compareAndSet(boolean, boolean)
-            return IS_VALID.compareAndSet(this, VALID, INVALID);
+            return set(VALID_MASK);
         }
 
         @Override
         public boolean isValid() {
-            return isValid == VALID;
+            return isNotSet(VALID_MASK);
```

Comment:
If "valid mask not set" means "is valid" then maybe the `VALID_MASK` should be named `INVALID_MASK` instead?

Reply:
I see the original code was using `int VALID = 0;` which mean that `isValid` was checking that valid was `==  VALID`.

Reply:
Yes, but my comment still applies ;-)

Reply:
Yep, i was trying to recollect why I have chosen such misleading name eheh 
I will happily fix it, thanks 🙏